src/main/java/edu/bu/metcs673/trackr/controller/TrackrUserController.java

```diff
@@ -0,0 +1,91 @@
+package edu.bu.metcs673.trackr.controller;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import edu.bu.metcs673.trackr.api.GenericApiResponse;
+import edu.bu.metcs673.trackr.common.CommonConstants;
+import edu.bu.metcs673.trackr.domain.TrackrUser;
+import edu.bu.metcs673.trackr.security.JWTUtil;
+import edu.bu.metcs673.trackr.service.TrackrUserService;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Controller for Users Management. Contains a 'Create' API for new users to
+ * register with the application. Also contains an API which generates a new JWT
+ * token for the user to use for authentication.
+ * 
+ * @author Tim Flucker
+ *
+ */
+@Slf4j
+@Validated
+@RestController
+@RequestMapping("/api/v1/users")
+public class TrackrUserController {
```

Comment:
it was renamed to avoid a naming clash with spring-security "User" class.  the application User class was renamed and other classes related to that entity have been renamed to avoid confusion.